### PR TITLE
Completed almost all features. Missing upload images.

### PR DESCRIPTION
**Pull request by agonzalezengr**

### CODE REVIEW

File: server/controller/QandA.js [hunk 1]
```diff
@@ -26,31 +25,52 @@ module.exports = {
...
+    }
+    return axios.put(process.env.DB_API + '/qa/' + x + val + '/helpful', null, {
+      headers: {'Authorization': process.env.GIT_TOKEN}
+    })
   },
 
-  report: (endpoint, id) => {
-
+  report: (id) => {
+    console.log('report');
```
Comment (kjmoyer) [inline server/controller/QandA.js:69]: If these are working, can we take the console.logs out before merge

File: src/components/QandA/Modals/AnswerModal.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            <h4 className="modal-subtitle">{productName}: {question_body}</h4>
+          </div>
+          <div className="modal-body">
+            <form onSubmit={handleSubmit}>
+              {Username}
+              {Email}
+              {Answer}
+              {photos.length < 5 &&
+                <div>
+                  <label htmlFor="photos">Not yet implemented: Upload your photos</label> <br></br>
```
Comment (kjmoyer) [inline src/components/QandA/Modals/AnswerModal.jsx]: We should take out this text for the presentation

File: src/components/QandA/QandA.jsx [hunk 3]
```diff
@@ -122,142 +123,113 @@ function QandA({ prod }) {
...
-    <>
+  const showMoreQuestions = (
+    <Button onClick={moreAnsweredQuestions} key={'showMoreQs'}>
+      <b>MORE ANSWERED QUESTIONS</b>
+    </Button>
+  )
+
+  return (
+    <React.Fragment>
       <h2> {'QUESTIONS & ANSWERS'}</h2>
```
Comment (kjmoyer) [inline src/components/QandA/QandA.jsx]: Capital first letters only

File: src/components/QandA/QandA.jsx [hunk 4]
```diff
@@ -122,142 +123,113 @@ function QandA({ prod }) {
...
       <Sort>
         <Search>
+        <Icon src={MagGlass}/>
           <Input
-            src={MagGlass}
             type="Text"
+            value={search}
+            placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
```
Comment (kjmoyer) [inline src/components/QandA/QandA.jsx]: Same here, only capital first letters

File: src/components/QandA/QandA.jsx [hunk 5]
```diff
@@ -122,142 +123,113 @@ function QandA({ prod }) {
...
-          <Icon src={MagGlass}/>
+          ></Input>
         </Search>
+        <br/>
+        {showData.length > 0 ?
+          <p>Click on a question to view the answers</p> :
+          <React.Fragment>
+            <p>There are no questions for this product. Click "Add a Question" to add a question.</p>
+            <Button onClick={() => setShowQModel(true)}>
+              <b>ADD A QUESTION +</b>
```
Comment (kjmoyer) [inline src/components/QandA/QandA.jsx:184]: Capital first letters only

File: src/components/QandA/QandA.jsx [hunk 6]
```diff
@@ -122,142 +123,113 @@ function QandA({ prod }) {
...
+        <QuestionModal
+          key={productID}
+          productID={productID}
+          productName={productName}
+          onClose={() => setShowQModel(false)}
+          showQModel={showQModel}
+        />
+        {showData.length > 0 && (
+          <Button onClick={() => setShowQModel(true)}>
+            <b>ADD A QUESTION +</b>
```
Comment (kjmoyer) [inline src/components/QandA/QandA.jsx:221]: Change to only capital first letters

File: src/components/QandA/QuestionList.jsx [hunk 7]
```diff
@@ -1,3 +1,313 @@
...
+    </Button>
+  )
+  const hideAnswers = (
+    <Button onClick={() => {
+      handleShowingAnswers(false);
+      setLoading(false);
+      setAnswerCount(2);
+      setSeeMoreClicked(!seeMoreClicked);
+    }}>
+      <b>COLLAPSE ANSWERS</b>
```
Comment (kjmoyer) [inline src/components/QandA/QuestionList.jsx]: Only capital first letters

File: src/components/QandA/QuestionList.jsx [hunk 8]
```diff
@@ -1,3 +1,313 @@
...
+    } else {
+      server.put(`/qa/${qOrA}/helpful`, x)
+        .then(response => {
+          swal("Thank You", 'Your feedback has been noted.', "success")
+          return setStateVariable(true)
+        })
+        .catch(err => console.error(err))
+        .then(() => {
+          setRerender(rerender + 1),
+          console.log(rerender)
```
Comment (kjmoyer) [inline src/components/QandA/QuestionList.jsx]: Remove console.log

**Review by kjmoyer — CHANGES_REQUESTED** [review]
Most comments about small fixes to style to be consistent across the page. I see a lot of work on the helpful/report buttons, but it doesn't look like it's working? We should still merge for now, but hopefully it's a quick fix to get it working since it looks like you put time into it last night

**Review by agonzalezengr — COMMENTED** [review]
Hope i didn't miss anything.

**Review by vickiwong85 — APPROVED** [review]

**Pull request closed without merging**